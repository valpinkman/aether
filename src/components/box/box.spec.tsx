import React from 'react'
import { shallow } from 'enzyme'
import Box, { Container } from './box'

describe('components', () => {
  describe('<Box />', () => {
    it('should render', () => {
      const wrapper = shallow(<Box />)
      expect(wrapper).toMatchSnapshot()
    })

    it('should render children', () => {
      const Child = () => <span>test</span>
      const wrapper = shallow(
        <Box>
          <Child />
        </Box>
      )

      expect(wrapper.find(Child)).toHaveLength(1)
      expect(wrapper).toMatchSnapshot()
    })

    it('should accept `colors` props', () => {
      const wrapper = shallow(
        <Box color="text.shade100" bg="background.default" />
      )

      expect(wrapper).toHaveLength(1)
      expect(wrapper.prop('color')).toBe('text.shade100')
      expect(wrapper.prop('bg')).toBe('background.default')
      expect(wrapper).toMatchSnapshot()
    })

    it('should accept `space` props', () => {
      const wrapper = shallow(<Box margin={20} p="2" />)

      expect(wrapper).toHaveLength(1)
      expect(wrapper.prop('margin')).toBe(20)
      expect(wrapper.prop('p')).toBe('2')
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('<Container />', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<Container />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
