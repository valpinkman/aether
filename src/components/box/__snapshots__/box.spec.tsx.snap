// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <Box /> should accept \`colors\` props 1`] = `
<StyledComponent
  bg="background.default"
  color="text.shade100"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "

  display: flex;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`components <Box /> should accept \`space\` props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "

  display: flex;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  margin={20}
  p="2"
/>
`;

exports[`components <Box /> should render 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "

  display: flex;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`components <Box /> should render children 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "

  display: flex;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <Child />
</StyledComponent>
`;

exports[`components <Container /> renders correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [
        [Function],
      ],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
  ",
          [Function],
          "

  display: flex;
",
          "
  flex: 1;
  flex-direction: column;
  height: 100vh;
  max-width: 100%;
",
        ],
      },
      "displayName": "Styled(styled.div)",
      "foldedComponentIds": Array [
        "sc-bdVaJa",
      ],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
